File: 프로그래머스/2/42842. 카펫/카펫.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer={};
    int tmp = (brown-4)/2;
    
    for(int i = 1; i<=tmp/2;i++){
        if((tmp-i)*i == yellow) {
            answer.push_back(tmp-i+2);
            answer.push_back(i+2);
        }
    }
    
    return answer;
}

File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    res = []
    
    res = nums(yellow)
        
    for i in range(len(res)):       
        temp = res[i][1] * 2 + res[i][0] * 2 + 4
            
        if temp == brown:
            answer = (res[i][1]+2, res[i][0]+2)
            break
            
    return answer

# 약수 구하기
def nums(yellow):
    res = []
    for i in range(1,yellow+1):
        if yellow % i == 0:
            res.append((i, yellow//i))
    return res
    